discovery_engine/lib/src/domain/event_handler.dart

```diff
@@ -225,6 +225,12 @@ class EventHandler {
     final activeDataRepository = HiveActiveDocumentDataRepository();
     final activeSearchRepository = HiveActiveSearchRepository();
     final engineStateRepository = HiveEngineStateRepository();
+    final aiStateHolders = [
+      documentRepository,
+      activeDataRepository,
+      activeSearchRepository,
+      engineStateRepository
+    ];
```

Comment:
I believe that by grouping them in this way here it's less likely to create a situation where we forget the clear a bit of AI state.